src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -46,4 +48,13 @@ public List<Book> findAll() {
             throw new RuntimeException("Can`t get all Books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return Optional.ofNullable(entityManager.find(Book.class, id));
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Cant find book by id: " + id, e);
```

Comment:
Not really. EntityNotFoundException should be thrown if Optional is empty. In this place book can be present but something else had happened

Reply:
Chane for RuntimeException